huaweicloud/services/dds/resource_huaweicloud_dds_audit_log_policy.go

```diff
@@ -0,0 +1,303 @@
+		CreateContext: resourceDdsAuditLogPolicyCreate,
+		UpdateContext: resourceDdsAuditLogPolicyUpdate,
+		ReadContext:   resourceDdsAuditLogPolicyRead,
+		DeleteContext: resourceDdsAuditLogPolicyDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"instance_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: `Specifies the ID of the DDS instance.`,
+			},
+			"keep_days": {
+				Type:         schema.TypeInt,
+				Required:     true,
+				Description:  `Specifies the number of days for storing audit logs.`,
+				ValidateFunc: validation.IntBetween(7, 732),
+			},
+			"audit_scope": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: `Specifies the audit scope.`,
+			},
+			"audit_types": {
+				Type:        schema.TypeList,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Optional:    true,
+				Computed:    true,
+				Description: `Specifies the audit type.`,
+			},
+			"reserve_auditlogs": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: `Specifies whether the historical audit logs are retained when SQL audit is disabled.`,
+			},
+		},
+	}
+}
+
+func resourceDdsAuditLogPolicyCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	// createAuditLog: create DDS audit log
+	var (
+		createAuditLogPolicyHttpUrl = "v3/{project_id}/instances/{instance_id}/auditlog-policy"
+		createAuditLogPolicyProduct = "dds"
+	)
+	createAuditLogPolicyClient, err := cfg.NewServiceClient(createAuditLogPolicyProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating DDS Client: %s", err)
+	}
+
+	instanceId := d.Get("instance_id")
+	createAuditLogPolicyPath := createAuditLogPolicyClient.Endpoint + createAuditLogPolicyHttpUrl
+	createAuditLogPolicyPath = strings.ReplaceAll(createAuditLogPolicyPath, "{project_id}",
+		createAuditLogPolicyClient.ProjectID)
+	createAuditLogPolicyPath = strings.ReplaceAll(createAuditLogPolicyPath, "{instance_id}",
+		fmt.Sprintf("%v", instanceId))
+
+	createAuditLogPolicyOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		OkCodes: []int{
+			200,
+		},
+	}
+	createAuditLogPolicyOpt.JSONBody = utils.RemoveNil(buildCreateAuditLogPolicyBodyParams(d))
+	_, err = createAuditLogPolicyClient.Request("POST", createAuditLogPolicyPath, &createAuditLogPolicyOpt)
+	if err != nil {
+		return diag.Errorf("error creating DDS audit log policy: %s", err)
+	}
+
+	resourceId, err := uuid.GenerateUUID()
+	if err != nil {
+		return diag.Errorf("unable to generate ID: %s", err)
+	}
+	d.SetId(instanceId.(string) + "/" + resourceId)
```

Comment:
```suggestion
	d.SetId(instanceId.(string))
```
resourceId is never used

Reply:
done